src/plugins/code-push.ts

```diff
@@ -0,0 +1,512 @@
+   * Calling this function is required on the first run after an update. On every subsequent application run, calling this function is a noop.
+   * If using sync API, calling this function is not required since sync calls it internally.
+   *
+   * @param notifySucceeded Optional callback invoked if the plugin was successfully notified.
+   * @param notifyFailed Optional callback invoked in case of an error during notifying the plugin.
+   */
+  @Cordova()
+  static notifyApplicationReady(): Promise<void> {
+    return;
+  }
+
+  /**
+   * Reloads the application. If there is a pending update package installed using ON_NEXT_RESTART or ON_NEXT_RESUME modes, the update
+   * will be immediately visible to the user. Otherwise, calling this function will simply reload the current version of the application.
+   */
+  @Cordova()
+  static restartApplication(): Promise<void> {
+    return;
+  }
+
+  /**
+   * Convenience method for installing updates in one method call.
+   * This method is provided for simplicity, and its behavior can be replicated by using window.codePush.checkForUpdate(), RemotePackage's download() and LocalPackage's install() methods.
+   *
+   * The algorithm of this method is the following:
+   * - Checks for an update on the CodePush server.
+   * - If an update is available
+   *         - If the update is mandatory and the alertMessage is set in options, the user will be informed that the application will be updated to the latest version.
+   *           The update package will then be downloaded and applied.
+   *         - If the update is not mandatory and the confirmMessage is set in options, the user will be asked if they want to update to the latest version.
+   *           If they decline, the syncCallback will be invoked with SyncStatus.UPDATE_IGNORED.
+   *         - Otherwise, the update package will be downloaded and applied with no user interaction.
+   * - If no update is available on the server, or if a previously rolled back update is available and the ignoreFailedUpdates is set to true, the syncCallback will be invoked with the SyncStatus.UP_TO_DATE.
+   * - If an error occurs during checking for update, downloading or installing it, the syncCallback will be invoked with the SyncStatus.ERROR.
+   *
+   * @param syncCallback Optional callback to be called with the status of the sync operation.
+   *                     The callback will be called only once, and the possible statuses are defined by the SyncStatus enum.
+   * @param syncOptions Optional SyncOptions parameter configuring the behavior of the sync operation.
+   * @param downloadProgress Optional callback invoked during the download process. It is called several times with one DownloadProgress parameter.
+   *
+   */
+  static sync(syncCallback?: SuccessCallback<SyncStatus>, syncOptions?: SyncOptions, downloadProgress?: SuccessCallback<DownloadProgress>) {
```

Comment:
This can be rewritten to this:
```typescript
@Cordova({
  observable: true,
  successIndex: 0,
  errorIndex: 3 // we don't need this, so we set it to a value higher than # of args, merge with latest master for this to work
})
static sync(syncOptions?: SyncOptions, downloadProgress?: Function): Observable<any> { return; }
```

I think the syncCallback is the important callback so we are wrapping that, the progress callback can stay as an optional param.